src/minidump.rs

```diff
@@ -1513,6 +1557,200 @@ impl MinidumpAssertion {
     }
 }
 
+fn read_string_list(
+    all: &[u8],
+    location: &md::MINIDUMP_LOCATION_DESCRIPTOR,
+    endian: scroll::Endian,
+) -> Result<Vec<String>, Error> {
+    let data = location_slice(all, location).or(Err(Error::StreamReadFailure))?;
+    if data.is_empty() {
+        return Ok(Vec::new());
+    }
+
+    let mut offset = 0;
+
+    let count: u32 = data
+        .gread_with(&mut offset, endian)
+        .or(Err(Error::StreamReadFailure))?;
+
+    let mut strings = Vec::with_capacity(count as usize);
+    for _ in 0..count {
+        let rva: md::RVA = data
+            .gread_with(&mut offset, endian)
+            .or(Err(Error::StreamReadFailure))?;
+
+        let string = read_string_utf8(&mut (rva as usize), all, endian)
+            .or(Err(Error::StreamReadFailure))?
+            .to_owned();
+
+        strings.push(string);
+    }
+
+    Ok(strings)
+}
+
+fn read_simple_string_dictionary(
```

Comment:
Can this function be implemented in terms of the existing [`read_stream_list`](https://github.com/luser/rust-minidump/blob/0759d861d60374379388d587d59b1df645a8b056/src/minidump.rs#L631) function? That one does seem to have a little bit of special-casing for alignment but otherwise it reads a length-prefixed array of structs, which is what this does.

Reply:
It can, but that would mean an intermediate allocation of the vector of raw entries that can be avoided by streaming them through. 

Here ideally we'd make `read_stream_list` an iterator so that we can map the raw values, but I think this could be its own PR. Would you prefer me to refactor in this changeset?

Reply:
No need to do the refactoring here. If you have an idea of how to make that work feel free to fix it in a followup.